### PR TITLE
Updated the tutorials to include the Python and R versions

### PR DESCRIPTION
**Pull request by FrankJVA**
The content of the Python tutorials was also updated to the latest version. Also, all the images were changed from jpg to png per the team's decision.

### CODE REVIEW

File: H2O-3/introduction-to-machine-learning-with-h2o-3-classification/introduction-to-machine-learning-with-h2o-3-classification.md [hunk 1]
```diff
@@ -1,1201 +1,2173 @@
...
+![default-gbm-training-metrics-2](assets/default-gbm-training-metrics-2.png)
+
+
+**Validation metrics:**
+
+
+![default-gbm-validation-metrics-1](assets/default-gbm-validation-metrics-1.png)
+![default-gbm-validation-metrics-2](assets/default-gbm-validation-metrics-2.png)
+
+The training and validation AUC are **0.878** and **0.854,** respectively. And we obtained a training and validation F1 Scores are **0.343** and **0.30** We can see the that the AUC and F1 Score of our default GBM is better than the values we obtained from our GLM and RF. We will now make the predictions with the GBM model as well.
```
Comment (5675sp) [inline H2O-3/introduction-to-machine-learning-with-h2o-3-classification/introduction-to-machine-learning-with-h2o-3-classification.md:978]: add a period after 0.30

File: H2O-3/introduction-to-machine-learning-with-h2o-3-regression/introduction-to-machine-learning-with-h2o-3-regression.md [hunk 2]
```diff
@@ -1,819 +1,1498 @@
...
+~~~python
+print("Default XGB MAE: %.4f \nTuned XGB MAE:%.4f" % (default_xgb_per.mae(), tuned_xgb_per.mae()))
+~~~
+**Output:**
+~~~python
+Default XGB MAE: 0.3108 
+Tuned XGB MAE:0.2986
+~~~
+
+In the case of an interest rate decision, an MAE of 0.30 is good enough as it tells us that, on average, the model would predict a very close interest rate to what someone with “x” characteristics should get. However, for some companies that MAE might be too high, so it all depends on the application to decide whether or not our MAE results are satisfactory. We will see if the Deep Learning model can yield a lower RMSE and MAE.
```
Comment (5675sp) [inline H2O-3/introduction-to-machine-learning-with-h2o-3-regression/introduction-to-machine-learning-with-h2o-3-regression.md:802]: not sure what you are tryin to say with this sentence: 

"However, for some companies that MAE might be too high, so it all depends on the application to decide whether or not our MAE results are satisfactory."

Consider rewriting the sentence

File: H2O-3/introduction-to-machine-learning-with-h2o-3-classification/introduction-to-machine-learning-with-h2o-3-classification.md [hunk 3]
```diff
@@ -1,1201 +1,2173 @@
...
+~~~
+
+Now that we have our dataset, we will explore some concepts and then do some exploration of the data and prepare it for modeling.
+
+Please note that we use the name H2O-3 to refer to our software, but when we usually use the simplified name, H2O, when we refer to specific functions or algorithms.
+
+### References
+
+[1] Our dataset is a subset of the [Freddie Mac Single-Family Loan-Level Dataset.](http://www.freddiemac.com/research/datasets/sf_loanlevel_dataset.html) It contains about 500,000 rows and is about 80 MB.
+
```
[the remark this answers is not in the document]
Comment (FrankJVA) [reply H2O-3/introduction-to-machine-learning-with-h2o-3-classification/introduction-to-machine-learning-with-h2o-3-classification.md:149]: The "Task x - R Version" gets removed in the HTML file

File: H2O-3/introduction-to-machine-learning-with-h2o-3-classification/introduction-to-machine-learning-with-h2o-3-classification.md [hunk 4]
```diff
@@ -1,1201 +1,2173 @@
...
+We will now take a look at the first ten predictions of our model with the following command:
+
+~~~r
+# Make predictions on validation set using our GLM model
+h2o.predict(glm, valid) 
+~~~
+ 
+![r-default-glm-predictions](assets/r-default-glm-predictions.png)
+
+**Note:** if you want to see more predictions use the `h2o.head()` function, and setting the value of `n` to the desired value, as shown in the line of code above; it should allow you to view all the predictions on the validation set that you want. This can be done as follow:
```
[the remark this answers is not in the document]
Comment (FrankJVA) [reply H2O-3/introduction-to-machine-learning-with-h2o-3-classification/introduction-to-machine-learning-with-h2o-3-classification.md:671]: I'm referring to the code that I previously showed, not the actual output

File: H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md [hunk 5]
```diff
@@ -1,576 +1,938 @@
...
+
+Lastly, let’s make some predictions on our test set.
+
+~~~python
+aml.predict(test)
+~~~
+
+![automl-cl-preds](assets/automl-cl-preds.png)
+
+As we mentioned in the first tutorial, the predictions we get are based on a probability. In the frame above, we have a probability for **FALSE,**, and another one for **TRUE**. The prediction, **predict**, is based on the threshold that maximizes the F1 score. For example, the threshold that maximizes the F1 is about `0.1061`, meaning that if the probability of **TRUE** is greater than the threshold, the predicted label would be **TRUE.**
```
Comment (5675sp) [inline H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md:362]: the following has an extra comma, remove it: "**FALSE,**,"

File: H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md [hunk 6]
```diff
@@ -1,576 +1,938 @@
...
+
+~~~r
+# Identify predictors and response for the classification use case
+ignore <- c("DELINQUENT", "PREPAID", "PREPAYMENT_PENALTY_MORTGAGE_FLAG", "PRODUCT_TYPE")
+y <- "DELINQUENT"
+
+x <- setdiff(colnames(train), ignore)
+~~~
+
+Now we are ready to run AutoML. Below you can see some of the default parameters that we could change for AutoML
```
Comment (5675sp) [inline H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md:378]: add a colon at the end of this sentence: "Now we are ready to run AutoML. Below you can see some of the default parameters that we could change for AutoML"

File: H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md [hunk 7]
```diff
@@ -1,576 +1,938 @@
...
+                     sort_metric = "AUC")
+~~~
+
+The only required parameters for H2O's AutoML are, `y` `training_frame,` and `max_runtime_secs,` which let us train AutoML for ‘x’ amount of seconds and/or `max_models,` which would train a maximum number of models. Please note that  `max_runtime_secs` has a default value, while `max_models` does not. The seed is the usual parameter that we set for reproducibility purposes. We also need a project name because we will do both classification and regression with AutoML. Lastly, we are using the default number of folds for cross-validation.
+
+The second line of code has the parameters that we need in order to train our model. For now, we will just pass x, y, and the training frame. Please note that the parameter `x` is optional because if you were using all the columns in your dataset, you would not need to declare this parameter. The `leaderboard frame` can be used to score and rank models on the leaderboard, but we will use the validation scores to do so because we will check the performance of our models with the test set.
+
+Below is a list of optional parameters that the user could be set for H2O’s AutoML
```
Comment (5675sp) [inline H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md:402]: rewrite this: "Below is a list of optional parameters that the user could be set for H2O’s AutoML"

to this: "Below is a list of optional parameters that the user could set for H2O’s AutoML"

File: H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md [hunk 8]
```diff
@@ -1,576 +1,938 @@
...
+The leaderboard shows that the GBM models clearly dominated this task. We can retrieve the best model with the `model.leader` command; but, what if we wanted to get another model from our leaderboard? One way to do so is shown below:
+
+~~~python
+# Get model ids for all models in the AutoML Leaderboard
+model_ids = list(aml.leaderboard['model_id'].as_data_frame().iloc[:,0])
+
+# Get the top GBM model
+gbm = h2o.get_model([mid for mid in model_ids if "GBM" in mid][0])
+~~~
+Note that you would need to change the name `GBM`, in the code above, to the name of the model that you want, and that should retrieve the desired model. For example, if you wanted to get the best XGBoost in the leaderboard, you would need to make the following change (you do not need to run the following line of code, this is just an example)
```
Comment (5675sp) [inline H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md:542]: rewrite the following sentence is a bit confusing: "Note that you would need to change the name `GBM`, in the code above, to the name of the model that you want, and that should retrieve the desired model."

File: H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md [hunk 9]
```diff
@@ -1,576 +1,938 @@
...
+
+~~~r
+# View the AutoML Leaderboard
+lb <- h2o.get_leaderboard(aml_reg)
+h2o.head(lb, n = -1)
+~~~
+
+![r-automl-reg-leaderboard_1](assets/r-automl-reg-leaderboard_1.png)
+
+**Note:** In the table above, only the top ten models are shown. You can view the other models by clicking **Next** button in your table.
```
Comment (5675sp) [inline H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md:654]: in the following sentence remove the word button: "**Note:** In the table above, only the top ten models are shown. You can view the other models by clicking **Next** button in your table. "

File: H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md [hunk 10]
```diff
@@ -1,576 +1,938 @@
...
+xgb <- h2o.getModel(grep("XGBoost", model_ids, value = TRUE)[1])
+~~~
+
+You could retrieve any model you want by providing a model ID to the `getModel` function as follow:
+
+~~~r
+gbm <- h2o.getModel("desired_model_id")
+~~~
+
+Now that we have retrieved the best model, we can take a look at some of the parameters
```
Comment (5675sp) [inline H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md:677]: add a colon at the end of this sentence

File: H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md [hunk 11]
```diff
@@ -1,576 +1,938 @@
...
+In the model summary above, we can see some of the parameters of our model, the metrics on the training data, and also the metrics from the cross-validation, as well as a detailed cross-validation metrics summary. We can also look at scoring history. However, by looking at the RMSE and MAE in the picture above for both training and validation, we can see that the model was starting to overfit because the training error is much lower than the validation error. Lastly, we can see the variable importance table, which shows both relative_importance, as well as scaled_importance, and also the percentage. For this model, the most important variable is `SELLER_NAME`, meaning that for our model, knowing which bank is providing the loan is very important.  
+
+Now, let’s see how the leader from our AutoML performs on our test set.
+
+~~~r
+# Save the test performance of the model
+gbm_test_perf <- h2o.performance(gbm, test)
+~~~
+
+Now you can print the RMSE and the MAE
```
Comment (5675sp) [inline H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md:716]: add a colon at the end of this sentence

File: H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md [hunk 12]
```diff
@@ -1,576 +1,938 @@
...
+Here you will be able to see all the “tables” that we have printed in our Jupyter Notebook, for example, you will see the outputs of the `.head()` function that we used earlier. Also, you will see a frame with the predictions and all the frames which we have worked with. Look for the `loan_level_50k.hex` frame, this is the dataset that we imported at the beginning of the tutorial. In the picture below, you will see three frames highlighted, the first one is the dataset we imported, and the other two are the train and test sets. 
+
+![flow-frames](assets/flow-frames.png)
+
+To import a file into Flow, you would just need to run the following command in a new cell
+~~~
+importFiles ["https://s3.amazonaws.com/data.h2o.ai/DAI-Tutorials/loan_level_50k.csv"]
+~~~
+
+**Note:** For a more detailed guide on how to import a file into Flow, check out [Task 3 of Introduction to Machine Learning with H2O-3 - Part 1](https://training.h2o.ai/products/introduction-to-machine-learning-with-h2o-part-1)
```
Comment (5675sp) [inline H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md:780]: add a period at the end of this sentence

File: H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md [hunk 13]
```diff
@@ -1,576 +1,938 @@
...
+
+Now that we can easily find our train and test set, scroll up again to the **Assistance** panel and click on `runAutoML`.
+
+![flow-run-automl](assets/flow-run-automl.png)
+
+You can name your model `flow-automl.` For *Training Frame* choose `train.` For the *Response Column* select `DELINQUENT.` For the *Leaderboard Frame* select `test.` 
+
+![flow-cl-automl-1](assets/flow-cl-automl-1.png)
+
+Click on the *balance_classes* box. Now, For *Ignored Columns* Select
```
Comment (5675sp) [inline H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md:795]: "for" after now should be lower case

File: H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md [hunk 14]
```diff
@@ -1,576 +1,938 @@
...
+Change the *sort metric* to `RMSE.` Let’s use all the models, so make sure all the models are unchecked, or click **None** in the *exclude_algos* menu. 
+
+![flow-automl-reg-2](assets/flow-automl-reg-2.png)
+
+Set the *Seed* to `42,` change the number of *max_runtime_secs* to `900,` and change the *stopping_metric* to `RMSE.` 
+
+
+![flow-automl-reg-3](assets/flow-automl-reg-3.png)
+
+Lastly, leave the expert settings as default, and click on **Build Model**
```
Comment (5675sp) [inline H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md:882]: add a colon at the en doc the sentence

File: H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md [hunk 15]
```diff
@@ -1,576 +1,938 @@
...
+
+
+![flow-automl-build](assets/flow-automl-build.png)
+ 
+
+Once it’s done, click on **View** and you should be able to see the **Leaderboard** and the **Event Log**
```
Comment (5675sp) [inline H2O-3/introduction-to-machine-learning-with-h2o-3-automl/introduction-to-machine-learning-with-h2o-3-automl.md:888]: add a colon at the end of the sentence

**Review by 5675sp — APPROVED** [review]

**Pull request merged**